### ISSUE TITLE
Balance timer start event instance creation across partitions

### ISSUE THREAD

**Issue by miguelpires** [camunda/camunda#1808]
At the moment, a timer start event creates timers on the deployment partition to avoid doing the same work in multiple brokers. Since this approach pushes the entire load of workflows with timer start events to the deployment partition, it would be better to create timers on all partitions and, each time a timer triggers, create the workflow instance on a different partition in order to distribute the load evenly. One way of doing this would be to add a "currentRepetition" value to the TimerRecord and, whenever a timer triggers, create the workflow instance on the corresponding partition and ignore it on the remaining:

```
if (record.getCurrentRepetition() % NUM_PARTITIONS == THIS_PARTITION) {
  createWorkflowInstance();
}
// increment the current repetition when rescheduling the timer
```

One problem with this approach is that if the brokers' clocks start to drift too much from each other, they might start executing at times that look wrong from a user's perspective.

Depends on #1787

**Comment by npepinpe** [comment]
Is this still the case?

**Comment by saig0** [comment]
Yes. Workflow instances with a timer start event are only created on partition 1.

**Comment by npepinpe** [comment]
> One problem with this approach is that if the brokers' clocks start to drift too much from each other, they might start executing at times that look wrong from a user's perspective.

I imagine this is true as well if the clock of the current leader of partition 1 drifts too much? To be honest, I think we're very vulnerable to clock drift in that respect, so I wouldn't worry too much in the scope of this issue.
